### PR TITLE
Merge integer conversions from bitwise plugin into main

### PR DESCRIPTION
**Pull request by ncave**
Although the bitwise plugin is a nice example, these integer conversions really are basic functionality and belong in the main.

### CODE REVIEW

File: src/fable/Fable.Compiler/Replacements.fs [hunk 1]
```diff
@@ -127,19 +127,39 @@ module Util =
...
+            | Fable.Number kind -> 
+                match kind with
+                | Int8 -> Some "($0 + 0x80 & 0xFF) - 0x80"
+                | UInt8 -> Some "$0 & 0xFF"
+                | Int16 -> Some "($0 + 0x8000 & 0xFFFF) - 0x8000"
+                | UInt16 -> Some "$0 & 0xFFFF"
+                | Int32 -> Some "($0 + 0x80000000 >>> 0) - 0x80000000"
+                | UInt32 -> Some "$0 >>> 0"
+                | Int64 -> Some "Math.trunc($0)" // only 53-bit (still better than nothing)
+                | UInt64 -> Some "($0 > 0) ? Math.trunc($0) : ($0 >>> 0)" // 53-bit positive, 32-bit negative
```
Comment (alfonsogarciacaro) [inline src/fable/Fable.Compiler/Replacements.fs]: Care is needed when emitting inline JS with multiple arg references (like `$0` in this case): if `$0` is a function call, a template like `($0 > 0) ? Math.trunc($0) : ($0 >>> 0)` will call the function three times (see #198). These cases should be replaced by a function in fable-core or by a [let-wrapping](https://github.com/fable-compiler/Fable/commit/bc101df191d637dbe0eae062b57aece0273dbcc7).

**Comment by alfonsogarciacaro** [comment]
The bitwise plugin was separated from the core because it's probably not worth to wrap all number operations with bound checks, however I agree we can do it for explicit conversions. I just left a comment, could you please check it? It also be nice to have some tests.

**Comment by ncave** [comment]
- replaced multiple args pattern with a function
- added tests
- tried to minimize conversions

**Comment by alfonsogarciacaro** [comment]
Awesome, thank you!

**Pull request merged**